### PR TITLE
Docs: OIDC flow endpoints

### PR DESCRIPTION
**Pull request by fairclothjm**
This PR adds docs for the OIDC flow endpoint APIs:
- openid configuration
- keys
- token
- authorization
- userinfo

 The docs are listed under Secret Engines -> Identity -> OIDC Provider.

This git branch is cut from the `docs/oidc-provider-api` branch. It is dependent on PR: https://github.com/hashicorp/vault/pull/12924.

### CODE REVIEW

File: website/content/api-docs/secret/identity/oidc-provider.mdx [hunk 1]
```diff
@@ -240,4 +240,260 @@ $ curl \
...
+```json
+{
+  "code": "BDSc9kVYljxND93YpveBuJtSvguM3AWe",
+  "state": "af0ifjsldkj"
+}
+```
+
+## Token Endpoint
+
+Provides the [token endpoint](https://openid.net/specs/openid-connect-core-1_0.html#TokenEndpoint)
```
Comment (austingebauer) [inline website/content/api-docs/secret/identity/oidc-provider.mdx:401]: Suggest to capitalize like "Token Endpoint" similar to "Authorization Endpoint" above. It's a small detail, but I like that it matches up with how it's capitalized in the spec.

File: website/content/api-docs/secret/identity/oidc-provider.mdx [hunk 2]
```diff
@@ -240,4 +240,260 @@ $ curl \
...
+  "expires_in": 3600,
+  "id_token": "eyJhbGciOiJSUzI1NiIsImtpZCI6ImEzMjk5ZWVmLTllNDEtOGNiYS1kNWExLTZmZWM2NjIyODRjYyJ9.eyJhdF9oYXNoIjoiMUdlQlEzUFdtUjJ2ajZVU2swSW42USIsImF1ZCI6InpTSktMVmk0R1BYS1o3TTZzUUEwY3FNc05VaHNPYkVTIiwiY19oYXNoIjoiN09SOUszNmhNdllENzJkUkFLUHhNdyIsImNvbnRhY3QiOnsiZW1haWwiOiJ2YXVsdEBoYXNoaWNvcnAuY29tIiwicGhvbmVfbnVtYmVyIjoiMTIzLTQ1Ni03ODkwIn0sImV4cCI6MTYzMzEwNjI5NCwiZ3JvdXBzIjpbImVuZ2luZWVyaW5nIl0sImlhdCI6MTYzMzEwNDQ5NCwiaXNzIjoiaHR0cDovLzEyNy4wLjAuMTo4MjAwL3YxL2lkZW50aXR5L29pZGMvcHJvdmlkZXIvbXktcHJvdmlkZXIiLCJuYW1lc3BhY2UiOiJyb290Iiwibm9uY2UiOiJhYmNkZWZnaGlqayIsInN1YiI6IjUwMDA3OTZlLTM2ZGYtMGQ4Yy02NDYwLTgxODUzZDliMjY2NyIsInVzZXJuYW1lIjoiZW5kLXVzZXIifQ.ehdLj6jnrJvltar1kkVSyNK48w2M5vkh5DTFJFZDqatnDWhQbbKGLZnVgd3wD6KPboXRaUwhGe4jDiTIiSoJaovOhsia77NKukym_ROLvGZw-LG7xaYkzJLnmEfeQhelLxWe0DHPROB7VXcFqBx8vX5hkuoVyqrB87vwiobK42pDPZ9MRsmbM2yzBC3wrnT7RQFtT4q2Bbyt9YIAHUaq9rU0PwJRoNISw6of1uQHo3_UzLdpwth7PEOEcI47OBGFA5vR_Gw3ocREfSrUWfCWOInAKCT43cImvg4Bts6qiZYfv9n-iNBq4AihGqq_VEF-hB1Hrprn7VgnEZ1VjUHaQQ",
+  "token_type": "Bearer"
+}
+```
+
+## UserInfo Endpoint
+
+Provides the [userinfo endpoint](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo)
```
Comment (austingebauer) [inline website/content/api-docs/secret/identity/oidc-provider.mdx:457]: Suggest to capitalize like "UserInfo Endpoint" similar to suggestion above.

File: website/content/api-docs/secret/identity/oidc-provider.mdx [hunk 3]
```diff
@@ -240,4 +240,260 @@ $ curl \
...
+provider's authorization endpoint.
+
+- `grant_type` `(string: <required>)` - The authorization grant type. The
+following grant types are supported: `authorization_code`.
+
+- `redirect_uri` `(string: <required>)` - The callback location where the
+authorization request was sent. This must match the `redirect_uri` used when the
+original authorization code was generated.
+
+### Headers
```
Comment (austingebauer) [inline website/content/api-docs/secret/identity/oidc-provider.mdx:424]: Nice! Like the headers sections.

**Review by austingebauer — APPROVED** [review]
Just a few minor suggestions. LGTM otherwise. Thanks!

File: website/content/api-docs/secret/identity/oidc-provider.mdx [hunk 4]
```diff
@@ -240,4 +240,260 @@ $ curl \
     --header "X-Vault-Token: ..." \
     --request DELETE \
     http://127.0.0.1:8200/v1/identity/oidc/scope/test-scope
-```
\ No newline at end of file
+```
+
+## Read .well-known Configurations
```
[the remark this answers is not in the document]
Comment (fairclothjm) [reply website/content/api-docs/secret/identity/oidc-provider.mdx:245]: Agreed

File: website/content/api-docs/secret/identity/oidc-provider.mdx (hunk 3 above)
Comment (fairclothjm) [reply website/content/api-docs/secret/identity/oidc-provider.mdx:422]: Thanks, I am thinking that they should be moved before the Parameters section.

**Comment by fairclothjm** [comment]
Hello @taoism4504 ! If you could please take a look at this PR it would be much appreciated. Thanks!

**Review by taoism4504 — APPROVED** [review]
LGTM

**Pull request merged**